src/main/java/org/folio/service/transformationfields/TransformationFieldsServiceImpl.java

```diff
@@ -52,6 +63,34 @@
     return promise.future();
   }
 
+  @Override
+  public void validateTransformations(List<Transformations> transformations) {
+    transformations.forEach(this::validateTransformation);
+  }
+
+  private void validateTransformation(Transformations transformation) throws ServiceException {
+    String fieldTransformation = transformation.getTransformation();
+    if(fieldTransformation.length() < TRANSFORMATION_MIN_ACCEPTABLE_LENGTH) {
+      throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, String.format(NOT_ACCEPTABLE_TRANSFORMATION, fieldTransformation));
+    }
+    String tag = fieldTransformation.substring(0, 3);
+    String ind1 = fieldTransformation.substring(3, 4);
+    String ind2 = fieldTransformation.substring(4, 5);
+    String subfield = fieldTransformation.substring(5);
+    if (!tag.matches("\\d{3}")) {
+      throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, String.format(INVALID_TAG_ERROR_MSG, tag));
+    }
+    if (!ind1.matches("[\\s\\d]")) {
```

Comment:
Here I split verification of each index to 2 "if statemanet" to have an error message which will point out the particular invalid index. If such granularity of error messaging isn't strict then it can be collapsed in one "if statemant" like:
if(!ind1.matches("[\\s\\d]") || !ind2.matches("[\\s\\d]"))
Your thoughts?